### ISSUE TITLE
Generate size relationships between non-contiguous objects in the perception generator

### ISSUE THREAD

**Issue by rgabbard** [isi-vista/adam#83]
This happens during the translation from the (curriculum-author-facing) situation representation to the (learner-facing) perceptual representation.

There seem to me to be three possible sources of size relationship information:

* size relationships explicitly specified in the situation representation (i.e. the curriculum author explicitly tells the system they want a giant box bigger than a person)
* relative sizes between kinds of objects encoded in the ontology (e.g. people are bigger than balls)
* relative sizes we can deduce from constraints on verbs (this information would also be stored in our ontology entries)

@denizbeser @mitchmarcus @charles-yang-upenn

**Comment by rgabbard** [comment]
@LichMaster98 :is this complete? Does the perception generator add size relationships for things not part of the same object schema?

**Comment by LichMaster98** [comment]
No this was waiting on https://github.com/isi-vista/adam/pull/185 Now that this has been merged in I am circling back to this issue before continuing onto other ones.

**Comment by rgabbard** [comment]
Re-opening until we confirm bullet 3 in the description is covered.

**Comment by rgabbard** [comment]
@LichMaster98 : tomorrow, check on the third bullet

**Comment by rgabbard** [comment]
Step 1: see if any of our action descriptions currently specify size relationships
Step 2: If so, make a test to be sure they are being translated to perceptions.

**Comment by LichMaster98** [comment]
The current implementation is too slow O(N^2) and needs to be improved. A possible solution is to store a prototypical size DiGraph on the Ontology. Then our situation builds its own DiGraph of explicit size relationships from given relationships and implicit size relations from our verbs. This could reduce our overall runtime by having us only calculate the prototypical size relationships of missing links between our DiGraph nodes.

Translating the DiGraph into our relationship structure would become a O(N) operation as we could just cascade down the graph's size relationships much like we currently construct our prototypical size relations.
